ecllibrary/std/Type.ecl

```diff
@@ -0,0 +1,241 @@
+      data readSerializedData(DATA meta, DATA parcel, unsigned4 fieldNum) : eclrtl,pure,library='eclrtl',entrypoint='readSerializedData';
+    end;
+    
+    /**
+     * Returns the binary type metadata structure for a record.
+     * 
+     * @return           A binary representation  of the type information
+     */
+    EXPORT DATA getBinaryTypeInfo() := outrec;
+     
+    EXPORT SET OF STRING fieldNames() := externals.getFieldNames(outrec);
+    EXPORT UNSIGNED4 numFields() := externals.getNumFields(outrec);
+    
+    /**
+     * Returns a serialized copy of a row.
+     * @param  inrow  Input row, in arbitrary layout. This will be translated to output layout then serialized.
+     * @return        Serialized data for the resulting output row
+     */
+    EXPORT DATA serialize(ROW inrow) := externals.serializeRow(outrec, inrow);
+
+    /**
+     * Functions to access individual fields within a serialized row without having to deserialize the whole row
+     * @param  parcel    Parcel to access. This should have been created using serialize() with matching output format, or results will be unpredictable.
+     * @param  fieldname Name of field to access.
+     */
+    EXPORT deserializedField(DATA parcel, STRING fieldname) := MODULE
+      EXPORT STRING   readString() := externals.readSerializedString(outrec, parcel, externals.getFieldNum(outrec, fieldname));;
+      EXPORT INTEGER  readInt() := externals.readSerializedInt(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT REAL8    readReal() := externals.readSerializedReal(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT BOOLEAN  readBool() := externals.readSerializedBool(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT UTF8     readUTF8() := externals.readSerializedUtf8(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+      EXPORT DATA     readData() := externals.readSerializedData(outrec, parcel, externals.getFieldNum(outrec, fieldname));
+    END;
+
+    /**
+     * Retrieve field name for a given field. Field indexes start at 1
+     * @param  fieldIndex  Field number to lookup
+     * @return             Corresponding field name, or blank if not found.
+     */
+    EXPORT STRING getFieldName(UNSIGNED fieldIndex) := externals.getFieldName(outrec, fieldIndex);
+    
+    /**
+     * Retrieve field index for a given field. Field indexes start at 1
+     * @param  fieldName   Field name to lookup
+     * @return             Corresponding field index, or 0 if not found.
+     */
+    EXPORT UNSIGNED getFieldNum(STRING fieldName) := externals.getFieldNum(outrec, fieldName);
+
+    /**
+     * Retrieve field type for a given field.
+     * @param  fieldName   Field name to lookup
+     * @return             Corresponding field type, or blank if not found.
+     */
+    EXPORT STRING getFieldType(STRING fieldName) := externals.getFieldType(outrec, getFieldNum(fieldName));
+
+  END;
+  
+
```

Comment:
Could add wrappers that use a non-module syntax here 

getFieldTypeFromVrec(retype, fieldname) := Types.vrec(retType).getFieldType(fieldname)